File: typescript-version/src/family/daughter.ts
import { Child } from './child';
import { Gender } from './person';
import { Father } from './father';
import { Mother } from './mother';

/**
 * Represents a daughter in the family
 */
export class Daughter extends Child {
  constructor(name: string, father: Father, mother: Mother) {
    super(name, father, mother, Gender.FEMALE);
  }

  /**
   * Describe the daughter's siblings
   */
  describeSiblings(): void {
    if (this.siblings.length === 0) {
      console.log(`${this.name} has no siblings.`);
    } else {
      for (const sibling of this.siblings) {
        if (sibling.gender === Gender.FEMALE) {
          console.log(`${this.name} is a ${this.getAge()}-year old sister`);
        } else {
          console.log(`${this.name} is a ${this.getAge()}-year old sister`);
        }
      }
    }
  }
}

File: python-version/family/daughter.py
"""
Daughter class - represents a daughter in the family
"""

from .child import Child
from .person import Gender
from .father import Father
from .mother import Mother


class Daughter(Child):
    """Represents a daughter in the family"""
    
    def __init__(self, name: str, father: Father, mother: Mother):
        """Initialize a daughter"""
        super().__init__(name, father, mother, Gender.FEMALE)
    
    def describe_siblings(self):
        """Describe the daughter's siblings"""
        if not self.siblings:
            print(f"{self.name} has no siblings.")
        else:
            for sibling in self.siblings:
                if sibling.gender == Gender.FEMALE:
                    print(f"{self.name} is a {self.get_age()}-year old sister")
                else:
                    print(f"{self.name} is a {self.get_age()}-year old sister")
